### PR TITLE
[Predictive Back] Add a new Predictive Back Samples folder.

### PR DESCRIPTION
**Pull request by ashnohe**
Set up recycler view to house interactive cards that will eventually show multiple samples. The recycler view does not currently have interactive samples. For now it describes how to view back-to-home and cross-activity animations.

https://photos.app.goo.gl/QGL7DrBtMbGVqTLk7

### CODE REVIEW

File: samples/user-interface/predictiveback/src/main/java/com/example/platform/ui/predictiveback/PBAnimation.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.platform.ui.predictiveback
+
+class PBAnimation(val title: String, val description: String)
```
Comment (marcelpinto) [inline samples/user-interface/predictiveback/src/main/java/com/example/platform/ui/predictiveback/PBAnimation.kt:19]: make it `data class` for just class holders

File: samples/user-interface/predictiveback/src/main/java/com/example/platform/ui/predictiveback/PBAnimation.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+val pbAnimations = mutableListOf<PBAnimation>().apply {
```
Comment (marcelpinto) [inline samples/user-interface/predictiveback/src/main/java/com/example/platform/ui/predictiveback/PBAnimation.kt:21]: it's better to use the constructor directly

```
mutableListOf(PBAnimation(...), PBAnimation(..), ...)
```

And if this list is not modified then defnie the variable type to be the immutable version List<PBAnimation>

File: samples/user-interface/predictiveback/src/main/java/com/example/platform/ui/predictiveback/PBViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.platform.ui.predictiveback
+
+import androidx.lifecycle.ViewModel
+
+class PBViewModel: ViewModel() {
```
Comment (marcelpinto) [inline samples/user-interface/predictiveback/src/main/java/com/example/platform/ui/predictiveback/PBViewModel.kt:21]: Just for this I would skip the VM. Just use the variable directly. It's simple and clear

File: samples/user-interface/predictiveback/src/main/java/com/example/platform/ui/predictiveback/PBHostingActivity.kt [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.example.platform.ui.predictiveback.databinding.HostingActivityBinding
+import com.google.android.catalog.framework.annotations.Sample
+
+@Sample(
+    name="Predictive Back Sample",
+    description="Shows Predictive Back animations."
+)
+class PBHostingActivity : AppCompatActivity() {
```
Comment (marcelpinto) [inline samples/user-interface/predictiveback/src/main/java/com/example/platform/ui/predictiveback/PBHostingActivity.kt:28]: Is the activity needed to demostrate the sample? If not maybe you could just use the fragment directly. This way you can skip the theme and all the boilerplate that comes with activity.

**Review by marcelpinto — CHANGES_REQUESTED** [review]
Are you planning to showcase it with compose too?

File: samples/user-interface/predictiveback/src/main/java/com/example/platform/ui/predictiveback/PBAnimation.kt (hunk 1 above)
Comment (ashnohe) [reply samples/user-interface/predictiveback/src/main/java/com/example/platform/ui/predictiveback/PBAnimation.kt:19]: acknowledged

File: samples/user-interface/predictiveback/src/main/java/com/example/platform/ui/predictiveback/PBAnimation.kt (hunk 2 above)
Comment (ashnohe) [reply samples/user-interface/predictiveback/src/main/java/com/example/platform/ui/predictiveback/PBAnimation.kt]: acknowledged

File: samples/user-interface/predictiveback/src/main/java/com/example/platform/ui/predictiveback/PBViewModel.kt (hunk 3 above)
Comment (ashnohe) [reply samples/user-interface/predictiveback/src/main/java/com/example/platform/ui/predictiveback/PBViewModel.kt]: acknowledged

File: samples/user-interface/predictiveback/src/main/java/com/example/platform/ui/predictiveback/PBHostingActivity.kt (hunk 4 above)
Comment (ashnohe) [reply samples/user-interface/predictiveback/src/main/java/com/example/platform/ui/predictiveback/PBHostingActivity.kt:28]: Yes, the activity is needed. From the sample page folks can swipe back to see the cross-activity animation. I'm also hoping to add a custom cross-activity animation.

**Comment by ashnohe** [comment]
Compose side of things tbd

**Review by marcelpinto — APPROVED** [review]

**Pull request merged**